File: Part 2 - Regression/Section 4 - Simple Linear Regression/Simple Linear Regression.py
# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Simple Linear Regression
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting results
y_pret = regressor.predict(X_test)

# Plotting Training stuff
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train , regressor.predict(X_train), color =  'blue')
plt.title('Salary vs Exp (Training Set)')
plt.xlabel('Experience')
plt.ylabel('Salary')
plt.show()

# Plotting Test Set
plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train , regressor.predict(X_train), color =  'blue')
plt.title('Salary vs Exp (Test Set)')
plt.xlabel('Experience')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.R
#importing dataset
dataset = read.csv('Salary_Data.csv')

library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Simple Linear Regression
regressor = lm(formula = Salary ~ YearsExperience ,
               data = training_set )

# Prediction
y_pred = predict(regressor, newdata = test_set)

# Plots
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point( aes(x = test_set$YearsExperience , y = test_set$Salary ),
              colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience , y = predict(regressor, newdata = training_set)) ,
            colour = 'blue') +
  ggtitle('Exp vs Salary (Test Set)') +
  xlab('Years Of Experience') + 
  ylab('Salary') 

